### PR TITLE
Implement Datumaroapp GUI for datumaro

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Add DatumaroApp
- Update documentation in following PR

### How to test

- Add unit tests in `tests\unit\gui`
- Some tests are failed due to copy `ContextVar` to thread, they will fix in the following PR

### Checklist

- [X] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [X] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: gui/datumaro_gui/components/multiple/streamlit_app.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+)
+from datumaro_gui.utils.drawing.css import box_style, custom_css
+from datumaro_gui.utils.readme import github_pypi_desc
+from streamlit import session_state as state
+
+from . import tabs
+
+
+def main():
+    print("__multiple__.main is called")
```
Comment (jihyeonyi) [inline gui/datumaro_gui/components/multiple/streamlit_app.py]: I think print codes are for debugging. How about using logger for this?

File: gui/datumaro_gui/components/single/tabs/export.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import os.path as osp
+
+import streamlit as st
+from datumaro_gui.utils.dataset.data_loader import DataRepo, SingleDatasetHelper
+from datumaro_gui.utils.dataset.state import get_download_folder_path
+from streamlit import session_state as state
+
+
+def main():
+    tasks = ["classification", "detection", "instance_segmentation", "segmentation", "landmark"]
```
Comment (jihyeonyi) [inline gui/datumaro_gui/components/single/tabs/export.py:15]: It seems that multiple and single export use same code a lot. 
I think this situation is similar in other tabs.
How about refactoring this later? Please make a JIRA Task for this purpose.

File: gui/datumaro_gui/components/multiple/streamlit_app.py (hunk 1 above)
Comment (sooahleex) [reply gui/datumaro_gui/components/multiple/streamlit_app.py]: Fix this [4bc52ea](https://github.com/openvinotoolkit/datumaro/pull/1279/commits/4bc52eaec15dafe601fb7f7605907135e9f08c5f)

File: gui/datumaro_gui/components/single/tabs/export.py (hunk 2 above)
Comment (sooahleex) [reply gui/datumaro_gui/components/single/tabs/export.py:15]: I will update this in following tasks. Thank you for letting me know this.

**Review by jihyeonyi — APPROVED** [review]
Let's add this first and refactor later. Looks good for now.

**Pull request merged**